frmts/hdf5/s102dataset.cpp

```diff
@@ -527,6 +578,8 @@ bool S102Dataset::OpenQualityOfSurvey(
 
     auto poDS =
         std::unique_ptr<GDALDataset>(poValuesArray->AsClassicDataset(1, 0));
+    if (!poDS)
+        return false;
```

Comment:
Maybe log an error here or is this nominal?

Reply:
this shouldn't happen normally. And if it did, AsClassicDataset() is supposed to have emitted an error message